src/main/java/ruslan/shastkiv/bookstore/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,69 @@
+package ruslan.shastkiv.bookstore.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import ruslan.shastkiv.bookstore.dto.cart.ShoppingCartDto;
+import ruslan.shastkiv.bookstore.dto.item.CartItemRequestDto;
+import ruslan.shastkiv.bookstore.dto.item.UpdateCartItemRequestDto;
+import ruslan.shastkiv.bookstore.service.cart.ShoppingCartService;
+import ruslan.shastkiv.bookstore.service.user.UserService;
+
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+@Tag(name = "Shopping Cart", description = "Endpoints for managing the shopping cart")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final UserService userService;
+
+    @GetMapping
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Get the shopping cart of the current user")
+    public ShoppingCartDto getShoppingCart(Authentication authentication) {
+        return shoppingCartService.getShoppingCart(userService.getUserId(authentication));
+    }
+
+    @PostMapping
+    @PreAuthorize("hasRole('USER')")
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Add a book to the shopping cart")
+    public ShoppingCartDto addBook(
+            Authentication authentication,
+            @RequestBody @Valid CartItemRequestDto requestDto) {
+        return shoppingCartService.addBookToCart(
+                userService.getUserId(authentication), requestDto);
+    }
+
+    @PutMapping("/items/{cartItemId}")
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Update quantity of a cart item")
+    public ShoppingCartDto updateQuantity(
+            @PathVariable Long cartItemId,
+            Authentication authentication,
+            @RequestBody @Valid UpdateCartItemRequestDto requestDto) {
+        return shoppingCartService.updateItemQuantity(
+                userService.getUserId(authentication), cartItemId, requestDto);
```

Comment:
In current implementation any user can change quantity of cart item for any other user

Reply:
I've tested the implementation, and everything works as expected. Users cannot modify cart items for other users.